src/components/AddingLayerForm/LayerSegmentsField/useInitSegmentsField.ts

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useState } from 'react';
+import { useField, useForm } from 'react-final-form';
+import { useNavigate } from 'react-router-dom';
+
+import api, { apiErrorHandler } from 'api';
+import { FieldName } from 'components/AddingLayerForm';
+import useErrorAlert from 'hooks/useErrorAlert';
+import { LayersListItem } from 'types/layer';
+import { LayerSegments, LayerSegmentsList, SegmentsFieldValue } from 'types/segment';
+
+const getInitialValue = (segmentsList: LayerSegmentsList): SegmentsFieldValue => {
+    return segmentsList.reduce<SegmentsFieldValue>((value, { id, title, roles, tags, activeState }) => {
+        value[`id-${id}`] = { id, title, roles, tags, activeState, isNew: false };
+        return value;
+    }, {});
+};
+
+const useInitSegmentsFieldValue = (): boolean => {
+    const {
+        input: { value },
+    } = useField<SegmentsFieldValue | null>(FieldName.Segments, { subscription: { value: true }, allowNull: true });
+    const {
+        input: { value: layer },
+    } = useField<LayersListItem | null>(FieldName.ParentLayer, { subscription: { value: true }, allowNull: true });
+    const [loading, setLoading] = useState(!value);
```

Comment:
а есть реальная зависимость от value?
кажется должно быть как-то так
```suggestion
    const [loading, setLoading] = useState(true);
```